### PR TITLE
Add logs for change of registration form settings

### PR DESCRIPTION
**Pull request by vtran99**
Request to add logs to action 'change of registration form settings'.

### CODE REVIEW

**Comment by ThiefMaster** [comment]
Maybe we can use this opportunity to add proper diff logging to registration form setting changes?

I don't like the idea of hardcoding a single case here and not logging anything else...

So basically, do something similar to what we do in `update_category` / `_log_category_update` or `_log_event_update`.

File: indico/modules/events/registration/operations.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                          data={'Changes': make_diff_log(changes, log_fields)})
+
+
+@make_interceptable
+def update_registration_form_settings(regform, data, *, skip=(), _extra_log_fields=None):
+    changes = regform.populate_from_dict(data, skip=skip)
+    db.session.flush()
+    signals.event.registration_form_edited.send(regform)
+    logger.info('Registration form %r updated with %r by %r', regform, data, session.user)
+    _log_registration_form_update(regform, changes, _extra_log_fields)
```
Comment (ThiefMaster) [inline indico/modules/events/registration/operations.py:53]:
```suggestion
    if changes:
        _log_registration_form_update(regform, changes, _extra_log_fields)
```

And remove the if in the other function

**Comment by vtran99** [comment]
I wonder if you hadn't make a mistake with the latest change for log title ?
Here are the 2 logs: 
the upper is with your change, the 2nd is mine, and it seems easier to filter logs on Registration Form with the old code...
image

**Comment by ThiefMaster** [comment]
Yes, it's consistent with the log messages for creation. The text there is the module, not the log subject (that's a mistake in my commit msg ;)). And if you view one of those messages, there is a button near the bottom of the modal to filter all log messages for that regform - so you see updates, creation, etc

**Pull request merged**